Cargo.toml

```diff
@@ -1,7 +1,7 @@
 [package]
 name = "hedwig"
 # TODO: When bumping to next major version, make sure to clean up the MRV and lints we allow in CI.
-version = "4.0.0"
+version = "4.0.1"
```

Comment:
As per [Semantic Versioning 2.0.0 Summary](https://semver.org/#summary):

> Given a version number MAJOR.MINOR.PATCH, increment the: [...] MINOR version when you add functionality in a backwards compatible manner, and

This PR adds new API surface, so the MINOR version field should be incremented.